src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,211 @@
+package com.commercetools.sync.commons.utils;
+
+import io.sphere.sdk.client.BlockingSphereClient;
+import io.sphere.sdk.client.QueueSphereClientDecorator;
+import io.sphere.sdk.client.RetrySphereClientDecorator;
+import io.sphere.sdk.client.SphereAccessTokenSupplier;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.client.SphereClientConfig;
+import io.sphere.sdk.http.AsyncHttpClientAdapter;
+import io.sphere.sdk.http.HttpClient;
+import io.sphere.sdk.retry.RetryAction;
+import io.sphere.sdk.retry.RetryContext;
+import io.sphere.sdk.retry.RetryPredicate;
+import io.sphere.sdk.retry.RetryRule;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClientConfig;
+
+import javax.annotation.Nonnull;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static java.lang.String.format;
+
+/**
+ * To create a Sphere Client with retry logic which computes a exponential backoff time delay in milliseconds.
+ * And handle all the configurations for the creation of client.
+ */
+public final class RetryableSphereClientWithExponentialBackoff {
+    protected static final long DEFAULT_MAX_DELAY = 60000;
+    protected static final long DEFAULT_INITIAL_RETRY_DELAY = 200;
+    protected static final int DEFAULT_MAX_RETRY_ATTEMPT = 5;
+    protected static final int DEFAULT_MAX_PARALLEL_REQUESTS = 20;
+    protected static final List<Integer> DEFAULT_STATUS_CODES_TO_RETRY =
+            new ArrayList(Arrays.asList(500, 502, 503, 504));
```

Comment:
Maybe Arrays.asList is enough. 
```suggestion
protected static final List<Integer> DEFAULT_STATUS_CODES_TO_RETRY = Arrays.asList(500, 502, 503, 504);
```

Reply:
Corrected. https://github.com/commercetools/commercetools-sync-java/pull/623/commits/40225ba32430c8c5c4f79f4677613585ccde6798